### ISSUE TITLE
RDAdmin, Manage Feeds: Crash when adding certain images to a feed.

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#841]
Hi.

There seems to be a problem with Magick image handling that leads to a crash of RDadmin while uploading an image if it determines the image being 'greyscale'
This can be very easily triggered by any 'dark on white' svg logo + title like image...

Here's the console dump... quite descriptive on the issue:

```
operador@tgnradio:~$ rdadmin
terminate called after throwing an instance of 'Magick::WarningCoder'
  what():  Magick: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `' @ warning/png.c/MagickPNGWarningHandler/1668
Aborted
```

For images containing 'evident' colorfullnes, with vivid reds, oranges, greens, blues, etc... there's no problem.

As a side note, I'm thinking whether this issue could break the process of apt-upgrading, since only 2 out of 13 images of my feeds have been fully migrated (thumbnails appearing on RDcatch, for instance) 
I'm wondering whether the generation of thumbnails, or even the completion of the upgrade to DB schema version, could have been broken during the upgrade at the very moment that Magick found the first 'grayscale' image.

Cheers.

**Comment by ElvishArtisan** [comment]
Vector graphics are definitely going to pose a problem. The [iTunes podcasting spec](https://help.apple.com/itc/podcasts_connect/#/itcb54353390) requires that images use JPEG or PNG encoding. We certainly do need to handle an attempt to import one better than by simply crashing though.

> As a side note, I'm thinking whether this issue could break the process of apt-upgrading, since only 2 out of 13 images of my feeds have been fully migrated (thumbnails appearing on RDcatch, for instance)

Very likely. Change those images to JPEG or PNG and I suspect the conversion will work fine.

**Comment by ElvishArtisan** [comment]
> The problem that I see is that, for Magick, it may be the actual content of the image (in this case, lack of basic colours) and not the usual basic features such as format, file size, etc, what may cause a problem, and this could be hard to realise by the end user.

Alas, the problem is more fundamental than that. Vector graphic images such as SVG are simply **not suitable** for use in RSS feeds. Aside from the iTunes compatibility issues, such images cannot be unambiguously rendered on a raster graphics display without additional information --e.g. color maps, image size-- also being provided. If you want to continue to use those images in your feeds, you'll need to convert them to JPEG or PNG and then import that. (I've used the GIMP to do this sort of conversion myself).

**Comment by alexolivan** [comment]
> you'll need to convert them to JPEG or PNG and then import that.

Ahhhh ... I didn't explain myself correctly!
I'm not using .SVGs , I'm using 'vector like' images, I mean, not a picture... Here an example of what i mean:

![this_is_techno_logo_RSS](https://user-images.githubusercontent.com/1419896/200885193-2388dcb9-387d-4220-a602-c74bed441647.png)

This image causes the issue to appear, regardless of being a .png

**Comment by ElvishArtisan** [comment]
Ah. Got it.

Any chance that you could post the actual file so I could test with it?

**Comment by ElvishArtisan** [comment]
I've just committed code (0c139bf and 3227e34) that will validate that an image file is of a supported format before importation. This should also keep the schema modification from bombing out (although you may end up with no thumbnails for those already-imported images that are not compatible.

(FWIW, I've also tested the 'This is Techno' image above, which works fine, but I'm not sure how representitve the image as served by GitHub is of the actual data used in your earlier tests.

Please test!

**Comment by alexolivan** [comment]
> (FWIW, I've also tested the 'This is Techno' image above, which works fine, but I'm not sure how representitve the image as served by GitHub is of the actual data used in your earlier tests.

Ok... I've never tried this before, I don't know if it's gonna work, just trying:

[Troublesome_greyscale_examples.zip](https://github.com/ElvishArtisan/rivendell/files/9978483/Troublesome_greyscale_examples.zip)

If works, that is a zip file containing the 3 images that, for my case, triggered the greyscale issue.
I'm going to try to re-import them after applying your commits, so I will report back on the test results ASAP anyways, to close this issue.

**Comment by ElvishArtisan** [comment]
> Ok... I've never tried this before, I don't know if it's gonna work, just trying:
>
> [Troublesome_greyscale_examples.zip](https://github.com/ElvishArtisan/rivendell/files/9978483/Troublesome_greyscale_examples.zip)

Received zip file, hash as follows:
```
[fredg@elastigirl alexolivan]$ sha1sum Troublesome_greyscale_examples.zip
63269dc4c7165850cdb44fc0b5ea697622cef729  Troublesome_greyscale_examples.zip
```

Unpacks as expected.

**Comment by ElvishArtisan** [comment]
All three of the images import and get processed by ImageMagick without error. I've also test the schema 264=>365 conversion with ```rddbmgr(8)``` with them. That too works perfectly here.

@$%^&*(?!

So, what is different between our two environments? My tests have been on CentOS 7, with the following ImageMagick packages:

- ImageMagick-c++-devel-6.9.10.68-6.el7_9.x86_64
- ImageMagick-c++-6.9.10.68-6.el7_9.x86_64
- ImageMagick-6.9.10.68-6.el7_9.x86_64
- ImageMagick-devel-6.9.10.68-6.el7_9.x86_64

**Comment by ElvishArtisan** [comment]
'Kay, I think I've finally gotten to the bottom of this. I was able to reproduce the problem on Ubuntu 22.04 (ImageMagick 6.9.11.60). The problem was due to a warning about an unnecessary RGB colormap in files containing a single bit deep --i.e. monochrome -- bitmap. The newer version throws an exception when it encounters the warning.

Fixed in 030bc6a. ```rddbmgr(8)``` will now print a fairly obnoxious warning when it encounters these exceptions, but will otherwise complete the conversions without bombing out. 

Please test!

**Comment by alexolivan** [comment]
Confirm fixed.

Not only it caught the RGB problem (as I saw in console) but that didn't prevent the image to be added to the feed! So, apparently, this makes .png files as safe to use as .jpg, very nice!

Thank you. Closing :-)
